src/Microsoft.DotNet.XHarness.CLI/Android/AndroidTestCommand.cs

```diff
@@ -1,67 +1,121 @@
-            Console.WriteLine($"  Timeout: {_arguments.Timeout.TotalSeconds}s");
-
-            Console.WriteLine("Arguments to instrumentation:");
-
-            foreach (KeyValuePair<string, string> pair in _arguments.InstrumentationArguments)
-            {
-                Console.WriteLine($"  {pair.Key} = {pair.Value}");
-            }
-
-            return Task.FromResult(0);
-        }
-    }
-}
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.CLI.Common;
+using Microsoft.Extensions.Logging;
+using Mono.Options;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.XHarness.CLI.Android
+{
+    internal class AndroidTestCommand : TestCommand
+    {
+        private readonly AndroidTestCommandArguments _arguments = new AndroidTestCommandArguments();
+        protected override ITestCommandArguments TestArguments => _arguments;
+
+        public AndroidTestCommand(ILogger logger) : base(logger)
+        {
+            Options = new OptionSet() {
+                "usage: android test [OPTIONS]",
+                "",
+                "Executes tests on and Android device, waits up to a given timeout, then copies files off the device.",
+                { "arg=", "Argument to pass to the instrumentation, in form key=value", v =>
+                    {
+                        string[] argPair = v.Split('=');
+
+                        if (argPair.Length != 2)
+                        {
+                            Options.WriteOptionDescriptions(Console.Out);
+                            return;
+                        }
+                        else
+                        {
+                            _arguments.InstrumentationArguments.Add(argPair[0].Trim(), argPair[1].Trim());
+                        }
+                    }
+                },
+                { "instrumentation|i=", "If specified, attempt to run instrumentation with this name instead of the default for the supplied APK.",  v => _arguments.InstrumentationName = v},
+            };
+
+            foreach (var option in CommonOptions)
+            {
+                Options.Add(option);
+            }
+        }
+
+        protected override Task<int> InvokeInternal()
+        {
+            _log.LogDebug($"Android Test command called: App = {_arguments.AppPackagePath}{Environment.NewLine}Instrumentation Name = {_arguments.InstrumentationName}");
+            _log.LogDebug($"Output Directory:{_arguments.OutputDirectory}{Environment.NewLine}Working Directory = {_arguments.WorkingDirectory}{Environment.NewLine}Timeout = {_arguments.Timeout.TotalSeconds} seconds.");
+            _log.LogDebug("Arguments to instrumentation:");
+
+            if (File.Exists(_arguments.AppPackagePath))

```

Comment:
nit: I personally prefer inverting the `if` and returning early than deep nesting

Reply:
Done!